client/src/preview/route/digitaltwins/DigitalTwinsPreview.tsx

```diff
@@ -1,96 +1,76 @@
 
-export const createDTTab = (error: string | null): TabData[] =>
+interface DTTabProps {
+  newDigitalTwinName: string;
+  setNewDigitalTwinName: React.Dispatch<React.SetStateAction<string>>;
+}
+
+export const createDTTab = ({
+  newDigitalTwinName,
+  setNewDigitalTwinName,
+}: DTTabProps): TabData[] =>
   tabs
-    .filter((tab) => tab.label === 'Manage' || tab.label === 'Execute')
+    .filter(
+      (tab) =>
+        tab.label === 'Manage' ||
+        tab.label === 'Execute' ||
+        tab.label === 'Create',
+    )
     .map((tab) => ({
       label: tab.label,
-      body: (
-        <>
-          <Typography variant="body1">{tab.body}</Typography>
-          <AssetBoard tab={tab.label} error={error} />
-        </>
-      ),
+      body:
+        tab.label === 'Create' ? (
+          <>
+            <Typography variant="body1">{tab.body}</Typography>
+            <CreatePage
+              newDigitalTwinName={newDigitalTwinName}
+              setNewDigitalTwinName={setNewDigitalTwinName}
+            />
+          </>
+        ) : (
+          <>
+            <Typography variant="body1">{tab.body}</Typography>
+            <AssetBoard tab={tab.label} />
+          </>
+        ),
     }));
 
-export const fetchSubfolders = async (
-  gitlabInstance: GitlabInstance,
-  dispatch: ReturnType<typeof useDispatch>,
-  setError: React.Dispatch<React.SetStateAction<string | null>>,
-) => {
-  try {
-    await gitlabInstance.init();
-    if (gitlabInstance.projectId) {
-      const subfolders = await gitlabInstance.getDTSubfolders(
-        gitlabInstance.projectId,
-      );
-      dispatch(setAssets(subfolders));
-      return subfolders;
-    }
-    dispatch(setAssets([]));
-    return [];
-  } catch (error) {
-    setError('An error occurred');
-    return [];
-  }
-};
-
-export const createDigitalTwinsForAssets = async (
-  assets: Asset[],
-  dispatch: ReturnType<typeof useDispatch>,
-) => {
-  assets.forEach(async (asset) => {
-    const gitlabInstance = new GitlabInstance(
-      sessionStorage.getItem('username') || '',
-      getAuthority(),
-      sessionStorage.getItem('access_token') || '',
-    );
-    await gitlabInstance.init();
-    const digitalTwin = new DigitalTwin(asset.name, gitlabInstance);
-    await digitalTwin.getDescription();
-    dispatch(setDigitalTwin({ assetName: asset.name, digitalTwin }));
-  });
-};
-
 export const DTContent = () => {
-  const [error, setError] = useState<string | null>(null);
+  const [newDigitalTwinName, setNewDigitalTwinName] = React.useState('');
   const dispatch = useDispatch();
-  const gitlabInstance = new GitlabInstance(
-    sessionStorage.getItem('username') || '',
-    getAuthority(),
-    sessionStorage.getItem('access_token') || '',
-  );
 
   useEffect(() => {
-    fetchSubfolders(gitlabInstance, dispatch, setError).then((assets) => {
-      if (assets) {
-        createDigitalTwinsForAssets(assets, dispatch);
-      }
+    const defaultFiles = [
```

Comment:
This array and the method to iterate upon this can moved to either to `gitlabDigitalTwin.ts`

Reply:
It has been moved to `file.ts` in `preview/util`.